src/clr-angular/utils/drag-and-drop/providers/drag-handle-registrar.spec.ts

```diff
@@ -0,0 +1,107 @@
+            fixture.destroy();
+        });
+
+        it("should be able to register element as default handle on assignment", function() {
+            dragHandleRegistrar.defaultHandleEl = draggableEl;
+            expect(dragEventListener.draggableEl).toBe(draggableEl);
+            expect(draggableEl.hasListener).toBeTruthy();
+            expect(draggableEl.classList.contains("drag-handle")).toBeTruthy();
+        });
+
+        it("should be able to register custom element as handle", function() {
+            dragHandleRegistrar.registerCustomHandle(customHandleEl);
+
+            expect(dragHandleRegistrar.customHandle).toBe(customHandleEl);
+            expect(dragEventListener.draggableEl).toBe(customHandleEl);
+
+            expect(customHandleEl.hasListener).toBeTruthy();
+            expect(customHandleEl.classList.contains("drag-handle")).toBeTruthy();
+        });
+
+        it("should be able to register custom element as drag handle after default handle is set", function() {
+            dragHandleRegistrar.defaultHandleEl = draggableEl;
+            dragHandleRegistrar.registerCustomHandle(customHandleEl);
+            // Once custom handle gets registered, listeners and drag styles should be removed from default element.
+            expect(draggableEl.hasListener).toBeFalsy();
+            expect(draggableEl.classList.contains("drag-handle")).toBeFalsy();
+
+            expect(dragHandleRegistrar.customHandle).toBe(customHandleEl);
+            expect(dragEventListener.draggableEl).toBe(customHandleEl);
+
+            expect(customHandleEl.hasListener).toBeTruthy();
+            expect(customHandleEl.classList.contains("drag-handle")).toBeTruthy();
+        });
+
+        it("should be able to unregister custom handle", function() {
+            dragHandleRegistrar.registerCustomHandle(customHandleEl);
+            dragHandleRegistrar.unregisterCustomHandle();
+
+            expect(dragHandleRegistrar.customHandle).toBeUndefined();
+        });
+
+        it("should be able to unregister custom handle and fall back to default handle if default handle is set before custom handle",
+           function() {
+               dragHandleRegistrar.defaultHandleEl = draggableEl;
+               dragHandleRegistrar.registerCustomHandle(customHandleEl);
+               dragHandleRegistrar.unregisterCustomHandle();
+               expect(dragHandleRegistrar.customHandle).toBeUndefined();
+               expect(dragEventListener.draggableEl).toBe(draggableEl);
+               expect(draggableEl.hasListener).toBeTruthy();
+               expect(draggableEl.classList.contains("drag-handle")).toBeTruthy();
+           });
+
+        it("should be able to unregister custom handle and fall back to default handle if default handle is set after custom handle",
+           function() {
+               dragHandleRegistrar.registerCustomHandle(customHandleEl);
+               dragHandleRegistrar.defaultHandleEl = draggableEl;
```

Comment:
I can see one missing test here. If you have:
```typescript
dragHandleRegistrar.registerCustomHandle(customHandleEl);
dragHandleRegistrar.defaultHandleEl = draggableEl;
```
You never check that the default handle doesn't override the listeners. The implementation has an `if` which is correct, but you don't have the corresponding unit test.

Reply:
Indeed, I missed that test. Thanks! I added the test in now.